### DISCUSSION TITLE
How can I inject livewire component into a form?

### DISCUSSION THREAD

**Discussion by ariesdg** [filamentphp/filament#9515] [Help] [answered]
Would it be possible to inject a livewire components into an existing resource create page?

**Comment by leandrocfe** [comment]
Could you explain more about it?

**Reply by ariesdg** [reply]
I'd like to create a search input where the user can input a discount code. Once submitted, it will validate the discount code and return whether it exists or not. If it exists, it will apply the discount to the form. I'm open to any better approaches for this.

**Reply by leandrocfe** [reply, marked as answer]
maybe this?
```php
TextInput::make('discount_code')
->suffixAction(
    \Filament\Forms\Components\Actions\Action::make('validate')
        ->icon('heroicon-o-arrow-right-circle')
        ->link()
        ->action(function(\Filament\Forms\Get $get) {
            //$get('discount_code')
            //your logic here
        })
)
```
